src/web/components/Nav/ExpandedMenu/ExpandedMenuToggle/VeggieBurger.tsx

```diff
@@ -60,51 +50,67 @@ const veggieBurgerIcon = ({
     `;
 
     return css`
+        background-color: currentColor;
+        /*
+            IMPORTANT NOTE:
+            we need to specify the adjacent path to the a (current) tag
+            to apply styles to the nested tabs due to the face we use ~
+            to support NoJS
+        */
+        ${`#${navCheckboxId}:checked ~ label & {
+            background-color: transparent;
+        }`}
+
         top: 50%;
         right: 0;
         margin-top: -1px;
         margin-left: auto;
         margin-right: auto;
         ${lineStyles};
-        background-color: ${showExpandedMenu ? 'transparent' : 'currentColor'};
+
         :before {
             ${lineStyles};
             ${beforeAfterStyles};
-            ${showExpandedMenu
-                ? `top: 0;
-            transform: rotate(-45deg);
-            `
-                : 'top: -6px;'};
+            /*
+                IMPORTANT NOTE:
+                we need to specify the adjacent path to the a (current) tag
+                to apply styles to the nested tabs due to the face we use ~
+                to support NoJS
+            */
+            top: -6px;
+            ${`#${navCheckboxId}:checked ~ label & {
+                top: 0;
+                transform: rotate(-45deg);
+            }`}
         }
         :after {
             ${lineStyles};
             ${beforeAfterStyles};
-            ${showExpandedMenu
-                ? `bottom: 0;
-            transform: rotate(45deg);
-            `
-                : 'bottom: -6px;'};
+            /*
+                IMPORTANT NOTE:
+                we need to specify the adjacent path to the a (current) tag
+                to apply styles to the nested tabs due to the face we use ~
+                to support NoJS
+            */
+            bottom: -6px;
+            ${`#${navCheckboxId}:checked ~ label & {
+                bottom: 0;
+                transform: rotate(45deg);
+            }`}
         }
     `;
 };
 
 export const VeggieBurger: React.FC<{
     display: Display;
-    toggleExpandedMenu: (value: boolean) => void;
-    showExpandedMenu: boolean;
-    ariaControls: string;
-}> = ({ display, toggleExpandedMenu, showExpandedMenu, ariaControls }) => {
+    navCheckboxId: string;
+}> = ({ display, navCheckboxId }) => {
     return (
-        <button
-            className={veggieBurger({ display, showExpandedMenu })}
-            onClick={() => toggleExpandedMenu(!showExpandedMenu)}
-            aria-controls={ariaControls}
-            aria-label="Toggle main menu"
-            data-link-name={`nav2 : veggie-burger : ${
-                showExpandedMenu ? 'hide' : 'show'
-            }`}
+        <NoJSButton
+            className={veggieBurgerStyles(display)}
+            navCheckboxId={navCheckboxId}
         >
-            <span className={veggieBurgerIcon({ showExpandedMenu })} />
-        </button>
+            <span className={veggieBurgerIconStyles(navCheckboxId)} />
```

Comment:
Is this an empty Label? It's likely we want to have some kind of text content in here for screenreaders, unless it's defined elsewhere?